### PR TITLE
rm matrix cache mgr [#183]

### PR DESCRIPTION
**Pull request by atolopko-czi**
#### Reviewers
**Functional:** 
@Bento007 

**Readability:** 
@danieljhegeman 

---

## Changes

The matrix data will be cached by CDN and user's browser so no need to cache ourselves at the server level. Cached items will be keyed by the S3 URL, so the data is immutable and can exist in CDN/browser caches for eternity.

- Remove all references to matrix data cache
- Remove caching config
- Remove cache classes, no longer used anywhere

### CODE REVIEW

File: server/data_common/data_adaptor.py [hunk 1]
```diff
@@ -122,6 +122,12 @@ def get_var_keys(self):
         # return list of keys
         pass
 
+    def __enter__(self):
+        return self
```
Comment (danieljhegeman) [inline server/data_common/data_adaptor.py:126]: this was a good little learning for me -- for some reason I always thought that `__enter__` had a default definition that returned self and that `__exit__` had a default definition that did nothing.

But apparently you had to define these, as I just learned from a little test on my own.

TIL 👍

File: server/default_config.py [hunk 2]
```diff
@@ -76,15 +76,6 @@
...
-    matrix_cache:
-      # The maximum number of datasets that may be opened at one time.  The least recently used dataset
-      # is evicted from the cache first.
-      max_datasets: 5
-
-      # A matrix is automatically removed from the cache after timelimit_s number of seconds.
-      # If timelimit_s is set to None, then there is no time limit.
-      timelimit_s: 30
-
   single_dataset:
```
Comment (danieljhegeman) [inline server/default_config.py:79]: we can convene later on removing everything related to single dataset mode

Comment (danieljhegeman) [reply server/default_config.py:79]: good cleanup task

File: server/data_common/data_adaptor.py (hunk 1 above)
Comment (atolopko-czi) [reply server/data_common/data_adaptor.py:126]: yeah, also still adjusting to not having explicit interfaces and/or base classes for all the special methods Python supports.  https://en.wikipedia.org/wiki/Duck_typing

File: server/default_config.py (hunk 2 above)
Comment (atolopko-czi) [reply server/default_config.py:79]: I gotcha covered: https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-explorer/181

**Review by Bento007 — APPROVED** [review]
I love commits where it's more deletions than additions. LGTM

**Comment by danieljhegeman** [comment]
tests pass 👍

**Comment by Bento007** [comment]
Do these changes depend on any remaining external work?

**Review by danieljhegeman — APPROVED** [review]

**Comment by atolopko-czi** [comment]
> Do these changes depend on any remaining external work?

Can you clarify "external work"? But in case I understand, we do need to deploy this along with v0.3 API changes, since removal of the caches relies upon the having API callers supply the dataset artifact S3 URLs.

**Pull request merged**